### PR TITLE
a bugfix for #1139

### PR DESCRIPTION
**Pull request by guy-maurel**

### CODE REVIEW

File: src/newlines.cpp [hunk 1]
```diff
@@ -2612,6 +2612,16 @@ void newlines_cleanup_braces(bool first)
...
+         tmp = chunk_get_next_type(pc, CT_SPAREN_CLOSE, pc->level);
+         if (tmp != nullptr)
+         {
+            prev = chunk_get_prev(tmp);
+            if (prev != nullptr)
+            {
+               // Issue #1139
+               newline_iarf_pair(prev, tmp, cpd.settings[UO_nl_before_closing_paren].a);
+            }
+         }
```
Comment (CDanU) [inline src/newlines.cpp]: As far as I see the `nl_before_closing_paren` option is only working on CT_IF chunks? Unless you plan to extend this I'd rename that option to `nl_before_if_closing_paren`.

Comment (guy-maurel) [inline src/newlines.cpp]: no because I use it also with `else if`

Comment (CDanU) [inline src/newlines.cpp]: Yes but `nl_before_closing_paren` would imply that all closing parenthesis, including for example `for(;;)`, would be affected.

Comment (guy-maurel) [inline src/newlines.cpp]: yes... next day

File: tests/input/cpp/bug_1139.cpp [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+                        )
+               {
+                  set_chunk_type(next, CT_LABEL_COLON);
+               }
+else if ((tmp == nullptr) ||
+                        ((tmp->type != CT_NUMBER) &&
+                         (tmp->type != CT_SIZEOF) &&
+                         !(tmp->flags & (PCF_IN_STRUCT | PCF_IN_CLASS))) ||
+                        (tmp->type == CT_NEWLINE)
+                        )
```
Comment (CDanU) [inline tests/input/cpp/bug_1139.cpp]: You are not testing if force works correctly. Pleas make one branch with 
```cpp
... (tmp->type == CT_NEWLINE))
```
and the other with
```cpp
... (tmp->type == CT_NEWLINE)

)
```

File: tests/config/bug_1139-f.cfg [hunk 3]
```diff
@@ -0,0 +1,6 @@
+output_tab_size            = 3
+indent_columns             = 3
+indent_with_tabs           = 0
+sp_before_sparen           = force
```
Comment (CDanU) [inline tests/config/bug_1139-f.cfg]: All of those are not needed to test `nl_before_if_closing_paren`

**Pull request merged**